Add option for RX pin pullup and send whole buffer for received raw messages in IRMQTTServer.ino

matgoebl opened the pull request:
Little improvements to IRMQTTServer.ino
- Add option for RX pin pullup (I'm using a SFH5110-38 receiver, that works with 5V. So I added a diode from SFH output to ESP input as simple level shifter, but this solution needs a 3.3V pullup at ESP side. I saved the external pullup by activating the internal pullup.)
- Send whole buffer for received raw messages (I wanted to decode the IR messages for my flameless christmas candles.)

matgoebl committed:
Add option for RX pin pullup in IRMQTTServer.ino

matgoebl committed:
Add option for RX pin pullup in IRMQTTServer.ino

matgoebl committed:
Send whole buffer for received raw messages in IRMQTTServer.ino

matgoebl committed:
Send whole buffer for received raw messages in IRMQTTServer.ino

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -1283,6 +1291,20 @@ void loop(void) {
     lastIrReceivedTime = millis();
     lastIrReceived = String(capture.decode_type) + "," +
         resultToHexidecimal(&capture);
+    if (capture.decode_type == -1) {
```
replace `-1` with `UNKNOWN`

crankyoldgit replied:
Also, on second thoughts, as we are sending this to MQTT and other people may have implemented stuff using the old expected behaviour. Lets put this extra behaviour all behind some #define switch that the user needs to change.
e.g
`#define REPORT_RAW_UNKNOWN false`

or move it to after `mqtt_client.publish(MQTTrecv, lastIrReceived.c_str());` so it is displayed in HTML only.

The raw line can produce a lot of output, and unless people have a large `MQTT_MAX_PACKET_SIZE` size set, we could hit that limit.

matgoebl replied:
Fixed using the enum `UNKNOWN` - thanks for the hint.

Made sending the whole buffer optional with `#define REPORT_RAW_UNKNOWNS false` - good idea.

Added warnings to increase `MQTT_MAX_PACKET_SIZE` - very helpful.

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -1283,6 +1291,20 @@ void loop(void) {
         resultToHexidecimal(&capture);
+    if (capture.decode_type == -1) {
+      lastIrReceived += ";";
+      for (uint16_t i = 1; i < capture.rawlen; i++) {
```
You could/should probably replace all of this with `lastIrReceived += "; " + resultToSourceCode(&capture);`

Not exactly the same, but it will use code under unit tests and produce a result that suits a lot of people more.

matgoebl replied:
This would be nice, but `resultToSourceCode(&capture)` adds extra space I tried to save.

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -1112,6 +1117,9 @@ void setup(void) {
   offset = irsend.calibrate();
 #if IR_RX
+#if IR_RX_PULLUP
+  pinMode(IR_RX, INPUT_PULLUP);
```
(No action required) Hmm. We should probably add that as an option in the library.

matgoebl replied:
Sure, that would be even better. I was just afraid to break existing APIs.

crankyoldgit reviewed (changes requested):
Nice changes. Great idea. Thanks for the contribution.

Just a few nitpicks/suggestions (see review comments) and, the Travis Build seems to have had an intermittent failure. Can you please go here and restart it?
https://travis-ci.org/markszabo/IRremoteESP8266/builds/472783248 We need to have it pass successfully before we can merge etc.

matgoebl committed:
Make sending the whole buffer optional.
Add comments to increase MQTT_MAX_PACKET_SIZE.
Fix using the enum value for UNKNOWN type.

matgoebl committed:
Make sending the whole buffer optional.
Add comments to increase MQTT_MAX_PACKET_SIZE.
Fix using the enum value for UNKNOWN type.

crankyoldgit committed:
Fix travis config (#591)

Try to fix some travis issues.
- Upgrade to newer arduino package.
- Breaking changes because of esp8266 core v2.5
  - CpuFrequency -> xtal
  - FlashSize -> eesz

crankyoldgit commented:
Please update/rebase/pull etc to include PR #591 as it will fix the travis issue.

matgoebl committed:
Merge branch 'master' of github.com:matgoebl/IRremoteESP8266

matgoebl reviewed (commented):
Changes rebased on upstream/master.

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -26,7 +26,7 @@
  *     - WiFiManager (https://github.com/tzapu/WiFiManager) (Version >= 0.14)
  *     - PubSubClient (https://pubsubclient.knolleary.net/)
- *   o You MUST change <PubSubClient.h> to have the following (or larger) value:
+ *   o You MUST change <PubSubClient.h> to have the following (or larger, particularly for REPORT_RAW_UNKNOWNS) value:
```
Lines should be <= 80 characters long  [whitespace/line_length]

matgoebl replied:
Thanks for the hint - I wasn't able to spot it in the CI logs.
Line wrapped.

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -151,6 +151,7 @@
 // Change to 'true'/'false' if you do/don't want these features or functions.
 #define USE_STATIC_IP false  // Change to 'true' if you don't want to use DHCP.
 #define REPORT_UNKNOWNS false  // Report inbound IR messages that we don't know.
+#define REPORT_RAW_UNKNOWNS false  // Report the whole buffer. MQTT_MAX_PACKET_SIZE of 1024 or more recommended.
```
examples/IRMQTTServer/IRMQTTServer.ino:154:  Lines should be <= 80 characters long  [whitespace/line_length] [2]

matgoebl replied:
Line wrapped.

crankyoldgit reviewed (changes requested):
Fix those two style/lint issues and it should pass the travis checks and be good to go, and I'll merge it once done.

matgoebl committed:
Fixed too long lines with comments

matgoebl reviewed (commented):
Now CI is happy - thanks for your help!

crankyoldgit reviewed (approved):
LGTM. Thanks for all the minor changes.

crankyoldgit commented:
FYI, this has been included in the new v2.5.4 release of the library.
